### DISCUSSION TITLE
Do I need escape --?  capture the line contain the string "--glob"

### DISCUSSION THREAD

**Discussion by jianhe-fun** [BurntSushi/ripgrep#2290] [Q&A] [answered]
The following two command failed.
` rg --help | rg  "--glob"`

` rg --help | rg  '--glob'`

pretty sure there is an line in help section contain string "--glob". How can I capture that line?

**Comment by s-p-turner** [comment, marked as answer]
rg --help | rg "\\-\\-glob"

**Reply by BurntSushi** [reply]
That is one way. Although note that you only need to escape the first dash, so `rg --help | rg '\--glob'` will work. Another way is `rg --help | rg -e --glob`.

@jianheMark Note this bit from the man page and `rg --help`:

```
    <PATTERN>    
            A regular expression used for searching. To match a pattern beginning with a
            dash, use the -e/--regexp flag.
            
            For example, to search for the literal '-foo', you can use this flag:
            
                rg -e -foo
            
            You can also use the special '--' delimiter to indicate that no more flags
            will be provided. Namely, the following is equivalent to the above:
            
                rg -- -foo
```

**Reply by s-p-turner** [reply]
Other characters to be aware of are * (which signifies the matching of zero or more occurrences of any characters) and . (which signifies the matching of one occurrence of any character). You have to be wary of searching for strings containing literal * or . characters because the result might not match what you expect.

For example (incorrect if you're looking for a literal asterisk):

```
$ echo " SPT " | rg -e "SPT*"
1: SPT
```

One way to correct this would be to escape the asterisk:
`$ echo " SPT " | rg -e "SPT\*"`

A further example (again, incorrect if you're looking for a literal period):

```
$ echo " SPT " | rg -e "SPT."
1: SPT
```

One way to correct this would be to escape the period:

`$ echo " SPT " | rg -e "SPT\."`

**Reply by BurntSushi** [reply]
Yes, although that's somewhat of a separate issue than why `rg --glob` doesn't work. The issue with `rg --glob` is that the arg parser doesn't know the difference between a `-` that indicates a flag or a `-` that starts a pattern. That's why `-e --glob` works: `-e` signals to the arg parser that the next argument should be treated as a value for the `-e` flag, regardless of whether it starts with a dash. For example `rg '[-]-glob'` works. So does `rg -- --glob`, since `--` signals to the parser that all subsequent arguments are positional and not flags.

For the general issue of escaping regex meta characters, see also the `-F` flag.
